crypto/pkcs12/pkcs12.go

```diff
@@ -0,0 +1,40 @@
+package pkcs12
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+
+	"github.com/AGWA-forks/go-pkcs12"
+	"github.com/cloudflare/cfssl/log"
+)
+
+// ParseAndEncode takes key, certificate, and optional password
+// as []byte and parses them to get a suitable format to encode them
+func ParseAndEncode(key, cert, password []byte) {
+	certBlock, _ := pem.Decode(cert)
+	certBytes := certBlock.Bytes
+	certificate, _ := x509.ParseCertificate(certBytes)
+
+	block, _ := pem.Decode(key)
+	privKey := block.Bytes
+	if block.Type == "RSA PRIVATE KEY" {
+		pkcsKey, _ := x509.ParsePKCS1PrivateKey(privKey)
+		Encode(pkcsKey, certificate, password)
+	} else if block.Type == "EC PRIVATE KEY" {
+		ecKey, _ := x509.ParseECPrivateKey(privKey)
+		Encode(ecKey, certificate, password)
+	}
+}
+
+// Encode is called by ParseAndEncode with a key, certificate, and optional password
+// to call AGWA-forks's pkcs12 encode function and outputs the pkcs12 file as "outfile.p12"
+func Encode(privateKey interface{}, certificate *x509.Certificate, password []byte) {
+	var none []*x509.Certificate
+	pfxData, err := pkcs12.Encode(privateKey, certificate, none, password)
+
+	err = ioutil.WriteFile("outfile.p12", pfxData, 0644)
+	if err != nil {
+		log.Errorf("failed to write new out file: %v", err)
+	}
```

Comment:
@kisom originally thought exporting the PKCS12 key might not be a good idea since it's the binary data. Or is there a different way of doing it that I'm missing?